### DISCUSSION TITLE
Sound glitching in snapclient

### DISCUSSION THREAD

**Discussion by carsoni** [snapcast/snapcast#974] [Show and tell]
Just to pass on my particular experience - maybe this will help someone else

I was having some difficulty with the sound gltching due to "pMiniBuffer -> full". This was happening about once every 10-20 seconds on both my streams, MPD and Shairport 

As part of my original attempt to setup snapcast server to manage the MPD and Shairport streams I had created two pipe sources in snapserver.conf "/tmp/snapfifo" and "/tmp/snapshairport" and had redirected Shairport output to the latter pipe in shairport-sync.conf.

By reverting to the "airplay:///..." source in snapserver.conf and allowing snapcast to manage shairport directly the gltiching has stopped.

I still get the occasional glitch now but their frequency is more like once per song / once per 2 songs which is acceptable. I think I will try a dedicated VLAN to see if even these can be eliminated by reducing network latency

Regards
Ian Carson

Context:

* Multi-room audio setup
* Music server is running on Ubuntu 20.04 VM on TrueNAS12 - no audio hardware
* Music server is using MPD for music database and Cantata for testing
* Music server is running snapcast snapserver
* I have snapclient running on 2 x RPi3B+ with HiFiBerry AMP2, 
* I am using snapweb on windows and the snapcast client on IOS15 on iPhone13Pro
* I have and MPD client running on IOS
* MPD has an audio_ouput to the /tmp/snapfifo pipe
* Snap server uses /tmp/snapfifo as a source named "MPD"

**Comment by rindlerblabla** [comment]
What sample rate are you using? I had issues with another Hifiberry card (DAC+ ADC PRO) and 44100. 48000 works just fine.
